### ISSUE TITLE
Full-screen apps un-full-screen when spawning floating windows.

### ISSUE THREAD

**Issue by NicTanghe** [leftwm/leftwm#1025]
### Describe the bug

I fullschreen an ap with "ToggleFullScreen",
When the window spawns a new floating window the app goes of fullschreenmode.

### Expected behavior (optional)

The app stays fullschreen and its child window is ontop(at all times)
(there is also some weird issue where child windows of child windows stay behind the window od the first child)

### Steps to reproduce (optional)

1. Make app fullschreen
2. make it spawn a child

### OS / Distro

RHEL-9

### Additional System Information (optional)

```text
NVIDIA GPU 
ANMD CPU
```

### LeftWM Check

```bash
:: LeftWM version: 0.4.1
:: LeftWM git hash: NONE
:: Enabled features: default journald-log lefthk lefthk-core tracing-journald
:: Checking feature dependencies . . .
    -> journald-log OK
    -> lefthk OK
:: Loading configuration . . .
    -> Configuration loaded OK
:: Checking keybinds . . .
    -> All keybinds OK
:: Checking environment . . .
    -> Environment OK
:: Checking theme . . .
ERROR: Could not parse theme file: 2:16: Expected option
```

theme file,

```rust
(border_width: 0,
margin: [12, 12, 15, 15],
default_border_color: "#222222",
floating_border_color: "#555555",
focused_border_color: "#FF3333",
)
```

**Comment by BKSalman** [comment]
I think the floating windows shouldn't "un-fullscreen" the window when they are spawned (so this part could be fixed), but I think the fullscreen window should be on top of everything, including floating windows.

but there will be a new feature for maximized windows ([I'm working on it now](https://github.com/leftwm/leftwm/pull/1121)) that unlike fullscreen windows, won't take the reserved space for the dock and margins, but will take the windows space and will be the only window on screen, and will keep the floating windows on top.

is that fair enough?

**Comment by VuiMuich** [comment]
Fair enough, especially if you hide your bar (eww can do this for sure, polybar I believe so as well) and set margins to 0 you get basically a fake fullscreen where floating windows stay on top.

**Comment by BKSalman** [comment]
ok, fixing the un-fullscreen-ing is a pretty simple fix, so will create a separate PR for it

**Comment by VuiMuich** [comment]
so a new config option `unfullscreen_on_window_spawn`?

**Comment by BKSalman** [comment]
I think this could be a decent solution

**Comment by VuiMuich** [comment]
Yeah, I'd say so.
Maybe even better name `leave_fullscreen_on_window_spawn`

**Comment by BKSalman** [comment]
or maybe `exit_fullscreen_on_window_spawn` lol, idk

**Comment by NicTanghe** [comment]
Ok i`m looking around and seeing that this is probably the most desirable if having windows float on top of full-screen apps is imposable. Although having floating windows on top of BKsalams maximized window is probably even more desirable.

**Comment by BKSalman** [comment]
alright, will create a PR that adds this option some time soon!

**Comment by Eskaan** [comment]
I want to note that **any** newly spawned window forces all windows on that tag to exit fullscreen.

I would assume there to be some code that explicitly does this because until [a few commits ago](https://github.com/leftwm/leftwm/commit/020ce3ae8f407165a084aa52501e8bf45f6ca2f1), no window could be above a window in fullscreen.

**Comment by NicTanghe** [comment]
It seems the fullschreen window no longer tries to stay on top after going into windowed mode.
when i tried to test maximised windows using :
(command: ToggleMaximized, value: "", modifier: ["modkey", "Shift"], key: "f"),
my config file stopped being recognised.

**Comment by VuiMuich** [comment]
Can you please post what `leftwm-check` reports?

**Comment by NicTanghe** [comment]
Below is when i use the config that woks with no issues.
And uses ToogleFullSchreen instead of Toggle Maximized

Toggle Fulschreen:
```
~
❯ leftwm-check
:: LeftWM version: 0.4.1
:: LeftWM git hash: NONE
:: Enabled features: default journald-log lefthk lefthk-core tracing-journald
:: Checking feature dependencies . . .
    -> journald-log OK
    -> lefthk OK
:: Loading configuration . . .
    -> Configuration loaded OK
:: Checking keybinds . . .
ERROR: Multiple commands bound to key combination modkey + l:
    -> (List(["modkey"]), "l")
    -> FocusWorkspaceNext
Help: change one of the keybindings to something else.

:: Checking environment . . .
    -> Environment OK
:: Checking theme . . .
ERROR: Could not parse theme file: 2:16: Expected option
```


Toggle Maximised:

```
~
❯ leftwm-check
:: LeftWM version: 0.4.1
:: LeftWM git hash: NONE
:: Enabled features: default journald-log lefthk lefthk-core tracing-journald
:: Checking feature dependencies . . .
    -> journald-log OK
    -> lefthk OK
:: Loading configuration . . .
Configuration failed. Reason: 85:34: Unexpected variant named `ToggleMaximized`in enum `BaseCommand`, expected one of `Execute`, `CloseWindow`, `CloseAllOtherWindows`, `SwapTags`, `SoftReload`, `HardReload`, `AttachScratchPad`, `ReleaseScratchPad`, `NextScratchPadWindow`, `PrevScratchPadWindow`, `ToggleScratchPad`, `ToggleFullScreen`, `ToggleSticky`, `GotoTag`, `ReturnToLastTag`, `FloatingToTile`, `TileToFloating`, `ToggleFloating`, `MoveWindowUp`, `MoveWindowDown`, `MoveWindowTop`, `FocusNextTag`, `FocusPreviousTag`, `FocusWindow`, `FocusWindowUp`, `FocusWindowDown`, `FocusWindowTop`, `FocusWorkspaceNext`, `FocusWorkspacePrevious`, `MoveToTag`, `MoveWindowToNextTag`, `MoveWindowToPreviousTag`, `MoveToLastWorkspace`, `MoveWindowToNextWorkspace`, `MoveWindowToPreviousWorkspace`, `NextLayout`, `PreviousLayout`, `SetLayout`, `RotateTag`, `IncreaseMainWidth`, `DecreaseMainWidth`, `SetMarginMultiplier`, `UnloadTheme`, `LoadTheme` instead
:: Checking environment . . .
    -> Environment OK
:: Checking theme . . .
ERROR: Could not parse theme file: 2:16: Expected option
```

second line is :
```
// | |      / __)_                                   / __|_)
```

**Comment by VuiMuich** [comment]
Well `ToggleMaximized` is a `0.5.1` feature, you seem to be on `0.4.1`.
The first error points out you appear to have bound `Mod+l` with two commands, try searching for `"l"` to find both entries.
The last error is referring to the `theme.ron` not the `config.ron`, you are probably missing the `implicit_option` flag, see wiki or upgrade to newer version as this was fixed iirc in 0.4.2

**Comment by NicTanghe** [comment]
super weird i could swear i deleted the old simlinks and copy`d over the newly compiled files
Il look at it tmorrow.
